studio/src/mas-fragment-variations.js

```diff
@@ -0,0 +1,106 @@
+import { LitElement, html } from 'lit';
+import { FragmentStore } from './reactivity/fragment-store.js';
+import { Fragment } from './aem/fragment.js';
+import { styles } from './mas-fragment-variations.css.js';
+
+// Inject styles once
+let stylesInjected = false;
+function injectStyles() {
+    if (stylesInjected) return;
+    const styleElement = document.createElement('style');
+    styleElement.textContent = styles;
+    document.head.appendChild(styleElement);
+    stylesInjected = true;
+}
+
+class MasFragmentVariations extends LitElement {
+    static properties = {
+        fragment: { type: Object, attribute: false },
+        loading: { type: Boolean, attribute: false },
+    };
+
+    constructor() {
+        super();
+        this.fragment = null;
+        this.loading = false;
+        injectStyles();
+    }
+
+    createRenderRoot() {
+        return this;
+    }
+
+    get variations() {
+        return this.fragment?.listLocaleVariations() || [];
+    }
+
+    get hasVariations() {
+        return this.variations && Array.isArray(this.variations) && this.variations.length > 0;
+    }
+
+    renderVariations() {
+        if (this.loading) {
+            return html`
+                <div class="loading-container">
+                    <sp-progress-circle indeterminate size="l"></sp-progress-circle>
+                    <p>Loading variations...</p>
+                </div>
+            `;
+        }
+
+        if (!this.hasVariations) {
+            return html`<p>No locale variations found</p>`;
+        }
+
+        return html`
+            <sp-table size="m">
+                <sp-table-body>
+                    ${this.variations.map(
```

Comment:
ideally FragmentStore instances for variations should be initialized where they are retrieved from the reppository. I propose to add methods in mas-repository.js for variations and init the stores there.